overlays/nodejs/serverless/function.ts

```diff
@@ -41,61 +42,58 @@ export interface Context {
 export type Handler = (event: any, context: Context, callback: (error: any, result: any) => void) => any;
 
 /**
- * FunctionOptions provides configuration options for the serverless Function.
+ * FunctionOptions provides configuration options for the serverless Function.  It is effectively
+ * equivalent to [aws.lambda.FunctionArgs] except with a few important differences documented at the
+ * property level.  For example, [role] is an actual iam.Role instance, and not an ARN. Properties
+ * like [runtime] are now optional.  And some properties (like [code]) are entirely disallowed.
  */
-export interface FunctionOptions {
+export type FunctionOptions = Overwrite<lambda.FunctionArgs, {
```

Comment:
@lukehoban I tried to do this without a mapping type.  However, while i could add the "code:never" property, i couldnt' change a property (like 'runtime' from required to optional).  That said, i think this guy reads pretty clearly.  It's a 'FunctionArgs', just with those properties overwritten.